source/material_model/steinberger.cc

```diff
@@ -189,11 +194,14 @@ namespace aspect
       // For an explanation on this formula see the Steinberger & Calderwood 2006 paper
       const double vis_lateral_exp = -1.0*lateral_viscosity_lookup->lateral_viscosity(depth)*delta_temperature/(temperature*adiabatic_temperature);
       // Limit the lateral viscosity variation to a reasonable interval
+
       const double vis_lateral = std::max(std::min(std::exp(vis_lateral_exp),max_lateral_eta_variation),1/max_lateral_eta_variation);
 
       const double vis_radial = radial_viscosity_lookup->radial_viscosity(depth);
+      const double vis_compositional = MaterialUtilities::average_value (volume_fractions, prefactors, viscosity_averaging);
+
 
-      return std::max(std::min(vis_lateral * vis_radial,max_eta),min_eta);
+      return std::max(std::min(vis_lateral * vis_radial * vis_compositional,max_eta),min_eta);
```

Comment:
I confess I don't understand this equation, as it looks like viscosity values are being multiplied together. Would it be possible to add the equation here (or further above) in a comment, along with a brief description?

Reply:
I second @bobmyhill renaming scheme for the values